### PR TITLE
[RFC] vim-patch:7.4.1516,7.4.1521

### PR DESCRIPTION
**Pull request by jbradaric**
```
vim-patch:7.4.1516

Problem:    Cannot change file permissions.
Solution:   Add setfperm().
```

https://github.com/vim/vim/commit/8049253b96838b3584600e5ad229abad37a95b10

```
vim-patch:7.4.1521

Problem:    File permission test fails on MS-Windows.
Solution:   Expect a different permission.
```

https://github.com/vim/vim/commit/8322e1f06e8fa39a6bb790a7d8d7db5d7aff3366

### CODE REVIEW

File: test/functional/legacy/file_perm_spec.lua [hunk 1]
```diff
@@ -21,7 +21,11 @@ describe('Test getting and setting file permissions', function()
     eq(9, call('len', call('getfperm', tempfile)))
 
     eq(1, call('setfperm', tempfile, 'rwx------'))
-    eq('rwx------', call('getfperm', tempfile))
+    if helpers.os_name == 'windows' then
+      eq('rwx------', call('getfperm', tempfile))
```
Comment (justinmk) [inline test/functional/legacy/file_perm_spec.lua]: shouldn't it be `'rw-rw-rw-'` as in the original patch?

File: runtime/doc/eval.txt [hunk 2]
```diff
@@ -2051,6 +2051,7 @@ serverlist()			String	get a list of available servers
 setbufvar({expr}, {varname}, {val})	set {varname} in buffer {expr} to {val}
 setcharsearch({dict})		Dict	set character search from {dict}
 setcmdpos({pos})		Number	set cursor position in command-line
+setfperm({fname}, {mode}	Number	set {fname} file permissions to {mode}
```
Comment (justinmk) [inline runtime/doc/eval.txt]: missing closing paren

**Comment by justinmk** [comment]
Note: for Vim's "new style" tests, they don't need to be converted. Wiki was updated to mention this, though maybe it should be more prominent. https://github.com/neovim/neovim/wiki/Merging-patches-from-upstream-Vim#code-differences

File: test/functional/legacy/file_perm_spec.lua [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  it('file permissions', function()
+    eq('', call('getfperm', tempfile))
+    eq(0, call('setfperm', tempfile, 'r------'))
+
+    call('writefile', {'one'}, tempfile)
+    eq(9, call('len', call('getfperm', tempfile)))
+
+    eq(1, call('setfperm', tempfile, 'rwx------'))
+    if helpers.os_name == 'windows' then
+      eq('rwx------', call('getfperm', tempfile))
```
Comment (justinmk) [inline test/functional/legacy/file_perm_spec.lua]: shouldn't it be `'rw-rw-rw-'` as in the original patch?

**Comment by jbradaric** [comment]
@justinmk I didn't notice the part about not porting new style tests. Should I then remove my test and use the original one?

File: test/functional/legacy/file_perm_spec.lua (hunk 3 above)
Comment (jbradaric) [inline test/functional/legacy/file_perm_spec.lua]: Yes, it should :)

**Comment by justinmk** [comment]
No, just for future reference.

**Comment by justinmk** [comment]
Verified test. Thanks @jbradaric

**Pull request merged**